mocks/routes/preset.js

```diff
@@ -0,0 +1,153 @@
+    {
+    id: "get-presets", // id of the route
+    url: "/api/preset", // url in path-to-regexp format
+    method: "GET", // HTTP method
+    variants: [
+      {
+        id: "success", // id of the variant
+        type: "json", // variant type
+        options: {
+          status: 200,
+          body: presets
+        }
+      },
+    ]
+    },
+    {
+        id: "add-presets", // route id
+        url: "/api/preset", // url in express format
+        method: "POST", // HTTP method
+        variants: [
+          {
+            id: "success",
+            type: "middleware", // variant of type "middleware"
+            options: {
+              middleware: (req, res) => {
+                    presets.push(req.body.preset)
+                    res.status(201)
+                    res.send(presets)
+              },
+            },
+          }
+        ],
+      },
+      {
+        id: "get-one-preset", // route id
+        url: "/api/preset/:id", // url in express format
+        method: "GET", // HTTP method
+        variants: [
+          {
+            id: "success",
+            type: "middleware", // variant of type "middleware"
+            options: {
+              middleware: (req, res) => {
+                    let toReturn = presets.filter(preset=>preset.id==req.params.id)
+                    if (toReturn) {
+                        res.status(201)
+                        res.send(toReturn)
+                    } else {
+                        res.status(404)
+                        res.send()
+                    }
+              },
+            },
+          }
+        ],
+    },
+    {
+        id: "delete-one-preset", // route id
+        url: "/api/preset/:id", // url in express format
+        method: "DELETE", // HTTP method
+        variants: [
+          {
+            id: "success",
+            type: "middleware", // variant of type "middleware"
+            options: {
+              middleware: (req, res) => {
+                    let index=-1
+                    presets.forEach(p => {
+                        if (p.id == req.params.id) {
+                            index = presets.indexOf(p)
+                            presets.splice(index, 1)
+                            res.status(200)
+                            res.send(req.params.id)
+                        }
+                    });
+                    res.status(404)
+                    res.send()
+              },
+            },
+          }
+        ],
+    },
+    {
+        id: "update-one-preset", // route id
+        url: "/api/preset/:id", // url in express format
+        method: "PUT", // HTTP method
+        variants: [
+          {
+            id: "success",
+            type: "middleware", // variant of type "middleware"
+            options: {
+              middleware: (req, res) => {
+                presets.forEach(p => {
+                    if (p.id == req.params.id) {
+                        p = req.params.preset
```

Comment:
The updated preset object will be in the body and not in the parameters
```suggestion
                        p = req.body
```